### PR TITLE
[quant][doc] Adding a table comparing eager and fx graph mode

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50413 [quant][doc] Adding a table comparing eager and fx graph mode**

Summary:

Test Plan:
.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25886960](https://our.internmc.facebook.com/intern/diff/D25886960)

### CODE REVIEW

File: docs/source/quantization.rst [hunk 1]
```diff
@@ -84,7 +84,42 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
 
 Eager Mode Quantization is a beta feature. User needs to do fusion and specify where quantization and dequantization happens manually, also it only supports modules and not functionals.
 
-FX Graph Mode Quantization is a new automated quantization framework in PyTorch, and currently it's a prototype feature. It improves upon Eager Mode Quantization by adding support for functionals and automating the quantization process. Although people might need to refactor the model a bit to make the model compatible with FX Graph Mode Quantization (symbolically traceable with torch.fx).
+FX Graph Mode Quantization is a new automated quantization framework in PyTorch, and currently it's a prototype feature. It improves upon Eager Mode Quantization by adding support for functionals and automating the quantization process, although people might need to refactor the model to make the model compatible with FX Graph Mode Quantization (symbolically traceable with ``torch.fx``).
+
+The following table compares the differences between Eager Mode Quantization and FX Graph Mode Quantization:
+
++-----------------+-------------------+-------------------+
```
Comment (raghuramank100) [inline docs/source/quantization.rst:91]: Do we also want to call out that both methods support static/dynamic and QAT? Is Fx support for dynamic quant on par with that of eager?

Comment (jerryzh168) [reply docs/source/quantization.rst:91]: Sure, yeah FX support is on par with eager

File: docs/source/quantization.rst [hunk 2]
```diff
@@ -84,7 +84,58 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
...
+|Operator         |Manual             |Automatic          |
+|Fusion           |                   |                   |
++-----------------+-------------------+-------------------+
+|Quant/DeQuant    |Manual             |Automatic          |
+|Placement        |                   |                   |
++-----------------+-------------------+-------------------+
+|Quantizing       |Supported          |Supported          |
+|Modules          |                   |                   |
++-----------------+-------------------+-------------------+
+|Quantizing       |Need Refactor      |Automatic          |
```
Comment (vkuzo) [inline docs/source/quantization.rst:108]: optional: `Manual` instead of `Need Refactor`, to match the others?

File: docs/source/quantization.rst [hunk 3]
```diff
@@ -84,7 +84,58 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
...
+|                 |Quantiztion Aware  |Quantiztion Aware  |
+|                 |Training:          |Training:          |
+|                 |Static             |Static             |
++-----------------+-------------------+-------------------+
+|Expect to work on|Yes                |No (We'll integrate|
+|arbitrary model  |                   |FX Graph Mode      |
+|                 |                   |Quantization into  |
+|                 |                   |domain libraries   |
+|                 |                   |like torchvision)  |
++-----------------+-------------------+-------------------+
```
Comment (vkuzo) [inline docs/source/quantization.rst:129]: I could see people having questions about what `works on an arbitrary model` means, it's possible to interpret this as "quantization will always work on any model without an accuracy drop".  Should we either remove this line or provide some clarifying details?

File: docs/source/quantization.rst [hunk 4]
```diff
@@ -84,7 +84,58 @@ PyTorch provides two different modes of quantization: Eager Mode Quantization an
...
+|Input/Output     |``torch.nn.Module``|``torch.nn.Module``|
+|Model Type       |                   |(May need some     |
+|                 |                   |refactors to make  |
+|                 |                   |the model          |
+|                 |                   |compatible with FX |
+|                 |                   |Graph Mode         |
+|                 |                   |Quantization)      |
++-----------------+-------------------+-------------------+
```
Comment (vkuzo) [inline docs/source/quantization.rst:137]: optional: `symbolically traceable torch.nn.Module`?

**Review by vkuzo — APPROVED** [review]
awesome!

File: docs/source/quantization.rst (hunk 4 above)
Comment (jerryzh168) [reply docs/source/quantization.rst:137]: I want to avoid mentioning symbolic traceability actually
